rfc/rfc_001_stamping-txs.md

```diff
@@ -0,0 +1,117 @@
+# RFC 048: Stamping transactions
+
+## Changelog
+
+* 11-09-2019: Initial draft
+
+## Context
+
+Here I am going to explain how we can decouple transactions from the block by stamping them.
+Stamping a transaction simply means putting the previous block hash in the header of a transaction.
+A *stamped transaction* unlike normal transaction has life time and it's only valid for a certain time (TTL).
+If `N` is the number of blocks after the stamp (or block hash), the *stamped transaction* after N blocks will be invalid.
+By stamping transactions we don't need to save the transactions inside a block and only keeping the transaction hash is fair enough.
```

Comment:
This does not make sense to me.

Every syncing node requires the transaction data in order to check validity of state. (In fact they require this from genesis) These transactions must be kept around.

Once you've synced, you don't need to keep any of a past blocks data around, unless you want to help end-users or your fellow peers sync. Stamping transactions does not affect either of these.

If you want to help end users, query for transactions, then this seems like strictly more memory over the ux-node locally maintaining a map of tx hash -> (blocknum, tx data), and forgetting the rest of the block data.

Reply:
> In fact they require this from genesis

Not necessary. If we can have some check points and export state of blockchain on that point, I don't think validating all transactions are necessary. However nodes can start building blockchain from the genesis. Now bitcoin is a good experience for us. People tends to connect to a pool or even download the blockchain data rather than starting from genesis.